src/lib/moment/start-end-of.js

```diff
@@ -16,8 +17,8 @@ export function startOf (units) {
         case 'isoWeek':
         case 'day':
         case 'date':
-            this.hours(0);
-            /* falls through */
+            setTime(this, 0, 0, 0, 0);
+            break;
         case 'hour':
             this.minutes(0);
```

Comment:
True! I also need to benchmark the functions `setMinutes(minutes, seconds, milliseconds)` and `setSeconds(seconds, milliseconds)` to see if they are even more efficient.